File: 21558 - Nithin/week 8/986-interval-list-intersections/986-interval-list-intersections.cpp
class Solution {
public:
    void swap(int& x,int& y){
        int temp = x;
        x = y;
        y = temp;
    }
    
    vector<vector<int>> intervalIntersection(vector<vector<int>>& firstList, vector<vector<int>>& secondList){
        vector<vector<int>> inter;
        int m = firstList.size();
        int n = secondList.size();
         
        if(m == 0 || n == 0)
            return inter;
        
        int maxi, mini,i = 0,j = 0;
        
        while(i < m && j < n ){
            vector<int> in;
            maxi = min(firstList[i][1], secondList[j][1]);
            mini = max(firstList[i][0], secondList[j][0]);
            if(mini<=maxi){
                in.push_back(mini);
                in.push_back(maxi);
                inter.push_back(in);
            }
            if(maxi == firstList[i][1])   
                i++;
            else
                j++;
        }
        
        return inter;
            
    }
};

File: 21551-naveen-siva-kumar/week8/986_Interval List Intersections.py
class Solution {
public:
    vector<vector<int>> intervalIntersection(vector<vector<int>>& firstList, vector<vector<int>>& secondList) {
        int n = firstList.size();
        int m = secondList.size();
        int i=0, j=0;
        int front,back;
        vector<vector<int>> ans;
        while(i<n && j<m) {
                   
            front = max(firstList[i][0], secondList[j][0]); 
            back = min(firstList[i][1], secondList[j][1]);
            if(back-front >= 0) {
                ans.push_back({front, back});         
            }
            
            if(firstList[i][1] < secondList[j][1]) i++;
            else  j++;
			
        }
        return ans;
    }
};
